pallets/parachain-staking/src/lib.rs

```diff
@@ -0,0 +1,1870 @@
+            collator: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            let delegator = ensure_signed(origin)?;
+            Self::delegation_schedule_revoke(collator, delegator)
+        }
+
+        #[pallet::weight(<T as Config>::WeightInfo::delegator_bond_more())]
+        /// Bond more for delegators wrt a specific collator candidate.
+        pub fn delegator_bond_more(
+            origin: OriginFor<T>,
+            candidate: T::AccountId,
+            more: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let delegator = ensure_signed(origin)?;
+            ensure!(
+                !Self::delegation_request_revoke_exists(&candidate, &delegator),
+                Error::<T>::PendingDelegationRevoke
+            );
+            let mut state = <DelegatorState<T>>::get(&delegator).ok_or(Error::<T>::DelegatorDNE)?;
+            state.increase_delegation::<T>(candidate.clone(), more)?;
+            Ok(().into())
+        }
+
+        #[pallet::weight(<T as Config>::WeightInfo::schedule_delegator_bond_less())]
+        /// Request bond less for delegators wrt a specific collator candidate.
+        pub fn schedule_delegator_bond_less(
+            origin: OriginFor<T>,
+            candidate: T::AccountId,
+            less: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let delegator = ensure_signed(origin)?;
+            Self::delegation_schedule_bond_decrease(candidate, delegator, less)
+        }
+
+        #[pallet::weight(<T as Config>::WeightInfo::execute_delegator_bond_less())]
+        /// Execute pending request to change an existing delegation
+        pub fn execute_delegation_request(
+            origin: OriginFor<T>,
+            delegator: T::AccountId,
+            candidate: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?; // we may want to reward caller if caller != delegator
+            Self::delegation_execute_scheduled_request(candidate, delegator)
+        }
+
+        #[pallet::weight(<T as Config>::WeightInfo::cancel_delegator_bond_less())]
+        /// Cancel request to change an existing delegation.
+        pub fn cancel_delegation_request(
+            origin: OriginFor<T>,
+            candidate: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            let delegator = ensure_signed(origin)?;
+            Self::delegation_cancel_request(candidate, delegator)
+        }
+
+        /// Hotfix to remove existing empty entries for candidates that have left.
+        #[pallet::weight(
+			T::DbWeight::get().reads_writes(2 * candidates.len() as u64, candidates.len() as u64)
+		)]
+        pub fn hotfix_remove_delegation_requests_exited_candidates(
+            origin: OriginFor<T>,
+            candidates: Vec<T::AccountId>,
+        ) -> DispatchResult {
+            ensure_signed(origin)?;
+            ensure!(candidates.len() < 100, <Error<T>>::InsufficientBalance);
```

Comment:
If we set limit on this vec, why not to use BoundedVec

Reply:
good point, but it pointed me to the even better question of why this code is in here at all.
It's a hotfix for invalid state moonbeam had on their chain with older versions of the pallet.
We don't have that -> don't need this hotfix